packages/plugin-jest/src/plugin-jest.ts

```diff
@@ -353,9 +355,25 @@ function packageEntryMatcherMap({runtimeName, entries, fs}: Package) {
 
   for (const {name, root} of entries) {
     map[
-      name ? join(runtimeName, `${name}$`) : `${runtimeName}$`
+      name ? `^${join(runtimeName, name)}$` : `^${runtimeName}$`
     ] = fs.resolvePath(root);
   }
 
   return map;
 }
+
+export function liftRepeatedValueInJSONString(
```

Comment:
This approach is very clever, and I get the desire for it, but I am not sure if it's the best idea. It is creating JavaScript with regular expressions which feels a little dicey.

What do you think of reducing the size of these name mappers by only including packages a given project depends on in the mapper? This would make them significantly smaller in most cases, and technically helps avoid tests incorrectly depending on packages they don't declare as dependencies, but is more complex to implement and might have unexpected results when a dependency is added while test watch mode is running

Reply:
I can definitely split this PR up and give the config simplification a bit more work 👍

Reply:
After taking a stab at it, it seems that Jest requires not only the immediate dependencies of a project to be mapped, but also the dependencies' dependencies (and so on). Which makes things a little trickier. In order for us to create a minimal map for each package we might need to build some kind of internal dependency graph 😬 Will investigate further.